app/views/search/index.html.erb

```diff
@@ -142,7 +142,7 @@
 
       <% page_numbers_for_pagination(@search.page_count, @search.page).each do |p| %>
         <% if p.is_a?(Integer) %>
-          <%= link_to p, search_path(@search.to_param), class: (@search.page == p ? 'cur' : '') %>
+          <%= link_to p, search_path(@search.to_param.merge(page: p)), class: (@search.page == p ? 'cur' : '') %>
```

Comment:
`@search.to_param` always was `page: 1` because it was the value when it was instantiated. updating it with the current page is working now